app/controllers/assignments_controller.rb

```diff
@@ -0,0 +1,73 @@
+class AssignmentsController < ApplicationController
+  include MakeNotice
+
+  before_action :set_assignment, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @assignments = Assignment.all
+    authorize Assignment
+  end
+
+  def show; end
+
+  def new
+    @assignment = Assignment.new
+    authorize @assignment
+  end
+
+  def edit; end
+
+  def create
+    @assignment = Assignment.new(assignment_params)
+    @assignment.to_pdf
+    authorize @assignment
+    if @assignment.save
+      @assignment.client.state = Client::RESERVED
+      @assignment.client.save
+      @assignment.volunteer.state = Volunteer::ACTIVE
+      @assignment.volunteer.save
+      redirect_to assignments_url, make_notice
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @assignment.update(assignment_params)
+      redirect_to assignments_url, make_notice
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    client = @assignment.client
+    volunteer = @assignment.volunteer
+    @assignment.destroy
+    client.state = Client::REGISTERED
+    client.save
+    if Volunteer.without_clients.include?(volunteer)
+      volunteer.state = Volunteer::ACCEPTED
+      volunteer.save
+    end
+    redirect_to assignments_url, make_notice
+  end
+
+  def find_volunteer
```

Comment:
Maybe move to `ClientController`, so you would have nicer routes file (see also comment for routes.rb).